data/storeDEPRECATED/mongo/mongo.go

```diff
@@ -49,12 +49,15 @@ func NewStore(cfg *storeStructuredMongo.Config, config *goComMgo.Config, lgr log
 		return nil, err
 	}
 
-	bucketStore, err := NewMongoBucketStoreClient(config, lg)
+	bucketStore, err := NewMongoBucketStoreClient(enableBucketStore, config, lg)
```

Comment:
why passing enableBucketStore to the constructor? If it is disabled then just don't create it.

Reply:
to ease integration in the project, it avoid checking nil value everywhere

Reply:
There are already test everywhere it is used. Either `if store.BucketStore != nil {` or `if d.BucketStore.enable {`

Could be easily replaced by `if d.BucketStoreEnabled {` if `BucketStoreEnabled ` becomes an attribute of class `Stores`